Fix out-of-bounds memory accesses exposed by xccblat3 testcase

martin-frbg opened the pull request:
Fixes #601. As far as I can tell, the data in these locations should be zero in any case.

martin-frbg committed:
Fix out-of-bounds accesses where the data should be zero anyway

brada4 commented:
In longer run they become non-zero, looping benchmarks down could expose similar blunders much better.

martin-frbg commented:
I do not believe my "solution" anymore either, currently trying to find if/where/when a01 grows beyond a.

brada4 commented:
This at least assists with most visible case. I am pretty sure non-zero is not expected from behind the array.

martin-frbg commented:
My current thinking is that the "if" block that contains the potential bug serves no useful purpose at all. Even when it happens to address something within the source array the matching result appears to be  obtained by another iteration of the main loop anyway. At least I have not managed to come up with a testcase where the problematic code is both executed at all and required for a correct result.

martin-frbg committed:
Comment out a code block that performs out-of-bounds memory accesses

...and does not appear to be needed even when it stays within the bounds of the array
